tag-expressions/python/cucumber_tag_expressions/parser.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: UTF-8 -*-
+# pylint: disable=missing-docstring
+"""
+Provides parsing of boolean tag expressions.
+
+.. code-block:: python
+
+    expression = TagExpressionParser.parse("a and (b or not c)")
+    assert True == expression.evaluate(["a", "other"])
+    assert "( a and ( b or not (c) ) )" == str(expression)
+
+UNSUPPORTED:
+* Support special tags w/ escaped-parens: @reqid\(10\)
+"""
+
+from __future__ import absolute_import
+from enum import Enum
+from cucumber_tag_expressions.model import Literal, And, Or, Not, True_
+
+
+# -----------------------------------------------------------------------------
```

Comment:
Removing the import-line will break the code and make all tag-parser related tests fail. These symbols are needed in the parser module.
Otherwise, the empty whitespace lines follow PEP-8 coding conventions in Python.